ui/src/components/common/Wallet.tsx

```diff
@@ -55,25 +59,30 @@ type FormProps = {
         text: string;
         value: string;
     }[];
+    role: MarketRole;
 }
 
-const AllocationForm: React.FC<FormProps> = ({ depositCid, options }) => {
+const AllocationForm: React.FC<FormProps> = ({ depositCid, options, role }) => {
     const [ exchange, setExchange ] = useState('');
     const [ loading, setLoading ] = useState(false);
     const [ error, setError ] = useState<ErrorMessage>();
 
     const operator = useWellKnownParties().userAdminParty;
-    const investor = useParty();
+    const party = useParty();
     const ledger = useLedger();
 
     const handleDepositAllocation = async (event: React.FormEvent) => {
         event.preventDefault();
 
         setLoading(true);
         try {
-            const key = wrapDamlTuple([operator, investor]);
+            const key = wrapDamlTuple([operator, party]);
             const args = { depositCid, provider: exchange };
-            await ledger.exerciseByKey(Investor.Investor_AllocateToProvider, key, args);
+            if (role === MarketRole.InvestorRole) {
+                await ledger.exerciseByKey(Investor.Investor_AllocateToProvider, key, args);
+            } else if (role === MarketRole.BrokerRole) {
+                await ledger.exerciseByKey(Broker.Broker_AllocateToProvider, key, args)
+            }
```

Comment:
This is a nice pattern! My PR has different custom `CustodianRequestForm`s for the Exchange, Investor, and Issuer that are all the same apart from the exercised choice. 

I'll probably steal this approach :)